src/lib/common/sol-platform-linux-common.c

```diff
@@ -785,3 +800,208 @@ sol_platform_linux_uevent_unsubscribe(const char *action, const char *subsystem,
+    }
+
+    timer_ctx.watcher = sol_fd_add(timer_ctx.fd,
+        SOL_FD_FLAGS_IN, system_clock_changed, NULL);
+
+    if (!timer_ctx.watcher) {
+        r = -ENOMEM;
+        goto err_exit;
+    }
+
+    return 0;
+
+err_exit:
+    close(timer_ctx.fd);
+    timer_ctx.fd = -1;
+    return r;
+}
+
+int
+sol_platform_unregister_system_clock_monitor(void)
+{
+    if (!timer_ctx.watcher)
+        return 0;
+
+    sol_fd_del(timer_ctx.watcher);
+    close(timer_ctx.fd);
+    timer_ctx.watcher = NULL;
+    timer_ctx.fd = -1;
+    return 0;
+}
+
+const char *
+sol_platform_impl_get_timezone(void)
+{
+    ssize_t n;
+    size_t offset;
+
+    n = readlink("/etc/localtime", timezone_str, sizeof(timezone_str));
+    if (n < 0) {
+        SOL_WRN("Could not readlink /etc/localtime");
+        return NULL;
+    }
+    timezone_str[n] = '\0';
+    if (strstartswith(timezone_str, "../usr/share/zoneinfo/"))
+        offset = strlen("../usr/share/zoneinfo/");
+    else if (strstartswith(timezone_str, "/usr/share/zoneinfo/"))
+        offset = strlen("/usr/share/zoneinfo/");
+    else {
+        SOL_WRN("The timzone is not a link to /usr/share/zoneinfo/");
+        return NULL;
+    }
+    memmove(timezone_str, timezone_str + offset, strlen(timezone_str) - offset + 1);
+    return timezone_str;
+}
+
+const char *
+sol_platform_impl_get_locale(enum sol_platform_locale_category category)
+{
+    int ctype;
+
+    ctype = sol_platform_locale_to_c_category(category);
+    SOL_INT_CHECK(ctype, < 0, NULL);
+    return setlocale(ctype, NULL);
+}
+
+int
+sol_platform_impl_apply_locale(enum sol_platform_locale_category category, const char *locale)
+{
+    int ctype;
+    char *loc;
+
+    ctype = sol_platform_locale_to_c_category(category);
+    SOL_INT_CHECK(ctype, < 0, -EINVAL);
+    loc = setlocale(ctype, locale);
+    SOL_NULL_CHECK(loc, -EINVAL);
+    return 0;
+}
+
+static enum sol_platform_locale_category
+system_locale_to_sol_locale(const struct sol_str_slice loc)
+{
+    static const struct sol_str_table map[] = {
+        SOL_STR_TABLE_ITEM("LANG", SOL_PLATFORM_LOCALE_LANGUAGE),
+        SOL_STR_TABLE_ITEM("LC_ADDRESS", SOL_PLATFORM_LOCALE_ADDRESS),
+        SOL_STR_TABLE_ITEM("LC_COLLATE", SOL_PLATFORM_LOCALE_COLLATE),
+        SOL_STR_TABLE_ITEM("LC_CTYPE", SOL_PLATFORM_LOCALE_CTYPE),
+        SOL_STR_TABLE_ITEM("LC_IDENTIFICATION", SOL_PLATFORM_LOCALE_IDENTIFICATION),
+        SOL_STR_TABLE_ITEM("LC_MEASUREMENT", SOL_PLATFORM_LOCALE_MEASUREMENT),
+        SOL_STR_TABLE_ITEM("LC_MESSAGES", SOL_PLATFORM_LOCALE_MESSAGES),
+        SOL_STR_TABLE_ITEM("LC_MONETARY", SOL_PLATFORM_LOCALE_MONETARY),
+        SOL_STR_TABLE_ITEM("LC_NAME", SOL_PLATFORM_LOCALE_NAME),
+        SOL_STR_TABLE_ITEM("LC_NUMERIC", SOL_PLATFORM_LOCALE_NUMERIC),
+        SOL_STR_TABLE_ITEM("LC_PAPER", SOL_PLATFORM_LOCALE_PAPER),
+        SOL_STR_TABLE_ITEM("LC_TELEPHONE", SOL_PLATFORM_LOCALE_TELEPHONE),
+        SOL_STR_TABLE_ITEM("LC_TIME", SOL_PLATFORM_LOCALE_TIME),
+    };
```

Comment:
if the value is not find in the table it will return SOL_PLATFORM_LOCALE_UNKNOWN,
check the line below
"sol_str_table_lookup_fallback(map, loc, SOL_PLATFORM_LOCALE_UNKNOWN);"